### PR TITLE
Feature 13093 data protection update

### LINKED ISSUE #13093: Update Data Protection for certain Data Fields [5]
### Problem Description

In order to be inline with data protection regulations in Luxembourg, certain data fields have to be updated.  Some are applicable to SORMAS overall, while others are applicable only to Luxembourg (see details below).

### Proposed Change

Tabs from the data dictionary:

<hr/>

### Person

<hr/>

Current:
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Person.birthdateMM | birthdateMM | Number | - |

Proposed (for all servers and instances):
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Person.birthdateMM | birthdateMM | Number | Personal |

(Apply to all servers).

<hr/>

### Location

<hr/>

Current:
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Location.city | city | Text | personal |

Proposed (only for Luxembourg instances):
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Location.city | city | Text | - |

(Apply removal of data protection restrictions for this field, but only if the server locale is set to Luxembourg)

<hr/>

### Case

<hr/>

Current:
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| CaseData.healthConditions | healthConditions | HealthConditions | - |
| CaseData.postpartum | postpartum | YesNoUnknown | - |
| CaseData.trimester | trimester | Trimester | - |

Proposed (for all servers and instances):
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| CaseData.healthConditions | healthConditions | HealthConditions | sensitive |
| CaseData.postpartum | postpartum | YesNoUnknown | sensitive |
| CaseData.trimester | trimester | Trimester | sensitive |

(Apply to all servers).

<hr/>

### Task

<hr/>

Current:
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Task.creatorComment | creatorComment | Text | - |
| Task.assigneeReply | assigneeReply | Text | - |

Proposed (for all servers and instances):
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Task.creatorComment | creatorComment | Text | -  : Add GDPR notice (pop-up) |
| Task.assigneeReply | assigneeReply | Text | -  : Add GDPR notice (pop-up) |

(Apply to all servers).

<hr/>

### Event

<hr/>

Current:
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Event.srcFirstName | scrFirstName | Text | - |
| Event.srcLastName | scrLastName | Text | - |
| Event.srcTelNo | scrTelNo | Text | - |
| Event.srcEmail | scrEmail | Text | - |

Proposed (for all servers and instances):
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Event.srcFirstName | scrFirstName | Text | sensitive |
| Event.srcLastName | scrLastName | Text | sensitive |
| Event.srcTelNo | scrTelNo | Text | sensitive |
| Event.srcEmail | scrEmail | Text | sensitive |

(Apply to all servers).

<hr/>

### Action

<hr/>

Current:
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Action.description | description | Text | - |

Proposed (for all servers and instances):
| Field ID | Field | Type | Data protection |
|---|---|---|---|
| Action.description | description | Text | - : Add GDPR notice (pop-up) |
| Action.description | Comments on execution| Text | - : Add GDPR notice (pop-up) |

(Apply to all servers).

<hr/>

### Added Value/Benefit

Enhanced data protection and compliance.

### Acceptance Criteria

- [ ] All fields (listed above) have new data protection attributes applied.

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
Fixes #13093

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13170

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13170

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java [hunk 1]
```diff
@@ -346,6 +346,7 @@ public class CaseDataDto extends SormasToSormasShareableDto implements IsCase {
 	@Valid
 	@EmbeddedPersonalData
 	@EmbeddedSensitiveData
+	@SensitiveData
```
Comment (leventegal-she) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java:349]: Shouldn't this and the other fields be sensitive only for luxembourg servers?

File: sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java [hunk 2]
```diff
@@ -171,7 +172,29 @@ private static String getCaption(FieldData fieldData) {
...
-		return I18nProperties.getPrefixDescription(fieldData.getI18NPrefix(), fieldData.getField().getName(), "");
+		String prefixDescription = I18nProperties.getPrefixDescription(fieldData.getI18NPrefix(), fieldData.getField().getName(), "");
+
+		String[] excludedForCountries = null;
+
+		Field field = fieldData.getField();
+		if (field.getAnnotation(PersonalData.class) != null) {
+			excludedForCountries = field.getAnnotation(PersonalData.class).excludeForCountries();
+
+		} else {
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java:183]: please change to else if

File: sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java [hunk 3]
```diff
@@ -171,7 +172,29 @@ private static String getCaption(FieldData fieldData) {
 	}
 
 	private static String getDescription(FieldData fieldData) {
-		return I18nProperties.getPrefixDescription(fieldData.getI18NPrefix(), fieldData.getField().getName(), "");
+		String prefixDescription = I18nProperties.getPrefixDescription(fieldData.getI18NPrefix(), fieldData.getField().getName(), "");
```
Comment (leventegal-she) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java:175]: please rename to fieldDescription

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/RichTextAreaCustom.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package de.symeda.sormas.ui.utils;
+
+import com.vaadin.event.FieldEvents;
+import com.vaadin.v7.ui.RichTextArea;
+
+public class RichTextAreaCustom extends RichTextArea {
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/RichTextAreaCustom.java:6]: is this still needed?

File: sormas-ui/src/main/java/de/symeda/sormas/ui/clinicalcourse/HealthConditionsForm.java [hunk 5]
```diff
@@ -80,39 +85,45 @@ protected void addFields() {
...
-		otherConditions.setRows(6);
-		otherConditions.setDescription(
-			I18nProperties.getPrefixDescription(HealthConditionsDto.I18N_PREFIX, OTHER_CONDITIONS, "") + "\n"
-				+ I18nProperties.getDescription(Descriptions.descGdpr));
-
-		initializeVisibilitiesAndAllowedVisibilities();
-		initializeAccessAndAllowedAccesses();
-
-		FieldHelper.setVisibleWhen(getFieldGroup(), HIV_ART, HIV, Arrays.asList(YesNoUnknown.YES), true);
+		if (UiUtil.permitted(UserRight.SEE_SENSITIVE_DATA_IN_JURISDICTION)) {
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/clinicalcourse/HealthConditionsForm.java:88]: making this field inaccessible should be done similarly like you did  for the NullableOptionGroup field

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java (hunk 1 above)
Comment (leventegal-she) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java:349]: I remember we discussed on the planning that if we make the health conditions on the case sensitive, then it has to be sensitive on all other places it is used

File: sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java (hunk 3 above)
Comment (leventegal-she) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java:175]: The data protection dictionary is country disease and feature dependent so it should not have this extra description, but should not show personal or sensitive in the data protection colum

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java:349]: updated in VaccinationDto, CaseExportDto and ContactDto

File: sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/info/EntityColumn.java:183]: changed

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/RichTextAreaCustom.java (hunk 4 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/utils/RichTextAreaCustom.java:6]: no, it's not needed anymore. Removed.

File: sormas-ui/src/main/java/de/symeda/sormas/ui/AboutView.java [hunk 6]
```diff
@@ -348,8 +348,13 @@ private String getCustomDocumentsPath() {
 
 	public void attachDataProtectionDictionaryDownloader(AbstractComponent target) {
 		new FileDownloader(new StreamResource(() -> new DownloadUtil.DelayedInputStream((out) -> {
-			String documentPath = FacadeProvider.getInfoFacade().generateDataProtectionDictionary();
-			IOUtils.copy(Files.newInputStream(new File(documentPath).toPath()), out);
+			try {
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/AboutView.java:351]: please revert this

File: sormas-ui/src/main/java/de/symeda/sormas/ui/clinicalcourse/HealthConditionsForm.java [hunk 7]
```diff
@@ -126,4 +137,8 @@ protected <F extends Field> F addFieldToLayout(CustomLayout layout, String prope
 
 		return super.addFieldToLayout(layout, propertyId, field);
 	}
+
+	private boolean displayFieldsAllowed() {
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/clinicalcourse/HealthConditionsForm.java:141]: please use similar solution like you did for the NullableOptionGroup

**Review by leventegal-she — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13170

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**